robot_smach_states/src/robot_smach_states/world_model/get_show_map.py

```diff
@@ -0,0 +1,110 @@
+            rospy.logerr("Could not generate a map image")
+            return "failed"
+
+        mark_ids = self.mark_ids.resolve() if hasattr(self.mark_ids, "resolve") else self.mark_ids
+
+        if mark_ids is not None:
+            for mark_id in mark_ids:
+                e = self.robot.ed.get_entity(uuid=mark_id)
+                if e is None:
+                    rospy.logdebug(f"Could not get entity for uuid: '{mark_id}'")
+                    continue
+
+                vs_image_frame = floor_plan.map_pose.frame.Inverse() * e.pose.frame.p
+
+                px = int(vs_image_frame.x() * floor_plan.pixels_per_meter_width)
+                py = int(vs_image_frame.y() * floor_plan.pixels_per_meter_height)
+
+                cv2.circle(floor_plan.map, (px, py), 25, (0, 0, 255), -1)
+
+        plan_points = self.plan_points.resolve() if hasattr(self.plan_points, "resolve") else self.plan_points
+
+        if plan_points is not None:
+            for point in plan_points:
+
+                vs_image_frame = floor_plan.map_pose.frame.Inverse() * point
+
+                px = int(vs_image_frame.x() * floor_plan.pixels_per_meter_width)
+                py = int(vs_image_frame.y() * floor_plan.pixels_per_meter_height)
+
+                cv2.circle(floor_plan.map, (px, py), 1, (255, 0, 0), -1)
+
+        os.makedirs(os.path.expanduser("~/ed/maps"), exist_ok=True)
+        filename = os.path.expanduser(
+            "~/ed/maps/floor_plan-{}.png".format(datetime.now().strftime("%Y-%m-%d-%H-%M-%S"))
+        )
+        cv2.imwrite(filename, floor_plan.map)
+        self.filename_des.write(filename)
+        rospy.loginfo(f"Wrote image to {filename}")
+
+        return "created"
+
+
+class GetMapAndShow(smach.StateMachine):
+    def __init__(self, robot, entity_ids=None, mark_ids=None, plan_points=None, duration=30):
+        super().__init__(self, outcomes=["done", "failed"])
+
+        filename_des = VariableDesignator(resolve_type=str).writeable
+
+        with self:
+            smach.StateMachine.add(
+                "GET_MAP",
+                GetMap(robot, filename_des, entity_ids, mark_ids, plan_points),
+                transitions={"created": "SHOW_MAP", "failed": "failed"},
+            )
+
+            smach.StateMachine.add(
+                "SHOW_MAP",
+                ShowImage(robot, filename_des, duration),
+                transitions={"succeeded": "done", "failed": "failed"},
+            )
+
+
+if __name__ == "__main__":
+    from robot_skills import get_robot
+
+    rospy.init_node(os.path.splitext("test_" + os.path.basename(__file__))[0])
+    r = get_robot("hero")
+    # Works for impuls environment
+    entities_des = VariableDesignator(["dining_room"], resolve_type=[str])
+    mark_ids_des = VariableDesignator(["initial_pose", "salon_table"], resolve_type=[str])
+    plan_points_des = VariableDesignator(resolve_type=Vector)
+    sm = GetMapAndShow(r, entities_des, mark_ids_des, plan_points_des)
+    sm.execute()
```

Comment:
Should this be here because it is very specific? Or do we want to keep this seperatly as an example?

Reply:
I will move this to examples. Did we ever decide that we want all `__main__` removed from the library? If we did, we should write it down. So we don't forget it again.